### PR TITLE
Added flag to return Logstash output

### PR DESCRIPTION
**Pull request by breml**
With this flag it is possible to output the Logstash output
even if Logstsh does exit with 0 exit code.

### CODE REVIEW

File: logstash-filter-verifier.go [hunk 1]
```diff
@@ -77,7 +81,7 @@ func runTests(logstashPath string, tests []testcase.TestCase, configPaths []stri
 		}
 
 		result, err := p.Wait()
-		if err != nil {
+		if err != nil || *logstashOutput {
 			message := fmt.Sprintf("Error running Logstash: %s.", err)
```
Comment (magnusbaeck) [inline logstash-filter-verifier.go]: `message` should only start like this if err != nil. Example of what it looks like now:

```
Running tests in client-decoration.json...
Logstash output:
Error running Logstash: %!s(<nil>).
Process output:
Sending logstash logs to /tmp/638530082.

Log:
{:timestamp=>"2017-02-02T20:03:43.426000+0000", :message=>"Reading config file", :config_file=>"/tmp/346208537/50-http-access-filter.conf.j2", :level=>:debug, :file=>"logstash/config/loader.rb", :line=>"69", :method=>"local_config"}
...
```

Perhaps just start with an empty `message` if there's no error?

File: logstash-filter-verifier.go [hunk 2]
```diff
@@ -89,7 +93,10 @@ func runTests(logstashPath string, tests []testcase.TestCase, configPaths []stri
 			} else {
 				message += "\nThe process wrote nothing to its logfile."
 			}
-			return errors.New(message)
+			if err != nil {
+				return errors.New(message)
+			}
+			userError("Logstash output:\n%s", message)
```
Comment (magnusbaeck) [inline logstash-filter-verifier.go]: We can probably just output `message` here. As it currently stands it'll look like this (with a correction for my previous comment):

```
Running tests in client-decoration.json...
Logstash output:
Process output:
Sending logstash logs to /tmp/638530082.

Log:
{:timestamp=>"2017-02-02T20:03:43.426000+0000", :message=>"Reading config file", :config_file=>"/tmp/346208537/50-http-access-filter.conf.j2", :level=>:debug, :file=>"logstash/config/loader.rb", :line=>"69", :method=>"local_config"}
...
```

File: logstash-filter-verifier.go (hunk 1 above)
Comment (breml) [inline logstash-filter-verifier.go]: Done.

File: logstash-filter-verifier.go (hunk 2 above)
Comment (breml) [inline logstash-filter-verifier.go]: Done.

**Pull request merged**